.gitbook/assets/ReadonlyREST-api.yml

```diff
@@ -0,0 +1,209 @@
+              examples:
+                NoAclBlockWasMatchedForbiddenResponseExample:
+                  $ref: "#/components/examples/NoAclBlockWasMatchedForbiddenResponseExample"
+                ForbidBlockMatchedResponseExample:
+                  $ref: "#/components/examples/ForbidBlockMatchedResponseExample"
+                TestSettingsNotConfiguredForbiddenResponseExample:
+                  $ref: "#/components/examples/TestSettingsNotConfiguredForbiddenResponseExample"
+                ImpersonationNotSupportedResponseExample:
+                  $ref: "#/components/examples/ImpersonationNotSupportedResponseExample"
+                ImpersonationNotAllowedResponseExample:
+                  $ref: "#/components/examples/ImpersonationNotAllowedResponseExample"
+                RorIsNotStartedYetResponseExample:
+                  $ref: "#/components/examples/RorIsNotStartedYetResponseExample"
+                RorFailedToStartResponseExample:
+                  $ref: "#/components/examples/RorFailedToStartResponseExample"
+                CannotSaveClusterWideSettingsOnFreeLicense:
+                  $ref: "#/components/examples/CannotSaveClusterWideSettingsOnFreeLicense"
+
+components:
+  securitySchemes:
+    http:
+      type: http
+      scheme: basic
+
+  schemas:
+    ForbiddenRorResponse:
+      type: object
+      required:
+        - error
+        - status
+      properties:
+        error:
+          type: object
+          required:
+            - type
+            - reason
+            - due_to
+          properties:
+            type:
+              type: string
+              enum:
+                - forbidden_response
+            reason:
+              type: string
+              enum:
+                - forbidden
+            due_to:
+              type: array
+              description: |
+                It's an array of causes why ReadonlyREST's ACL forbade the request. 
+                WARNING: Please notice, that one element array ES used to print as a string!!!
+              items:
+                type: string
+                enum:
+                  - OPERATION_NOT_ALLOWED
+                  - FORBIDDEN_BY_BLOCK
+                  - TEST_SETTINGS_NOT_CONFIGURED
+                  - IMPERSONATION_NOT_SUPPORTED
+                  - IMPERSONATION_NOT_ALLOWED
+                  - READONLYREST_NOT_READY_YET
+                  - READONLYREST_NOT_ENABLED
+                  - READONLYREST_FAILED_TO_START
+                  - cannotSaveClusterWideSettingsOnFreeLicense
+        status:
+          type: number
+          example: 403
+
+    XRorCorrelationId:
+      type: string
+      description: "Correlation ID to correlate requests/responses logs within one ROR KBN session"
+      example: "CED054A0-9881-401E-AEED-4FB4C5915DD6"
+
+    UpdateMainSettingsRequest:
+      type: object
+      required:
+        - settings
+      properties:
+        settings:
```

Comment:
I wonder if we should do it in the same way we do in the ROR ES internal API. Here we pass JSON and in the value of the `settings` field, we pass escaped YAML. 

WDYT about getting rid of the JSON and using [`application/yaml`](https://www.ietf.org/archive/id/draft-ietf-httpapi-yaml-mediatypes-00.html) instead ?

Reply:
To be clear, do you mean sending yaml content, right? WDYT about letting the user also directly upload yaml file?

Reply:
As long as we are sure we preserve spaces, newlines, comments, and yaml formatting (e.g. arrays as lists vs arrays in square brackets) in their natural position as the user typed them (including errors).

Reply:
yes, the whole body should be ROR YAML. And we can provide a second endpoint with file upload. But uploading a file is not always easy to do (due to HTTP clients' impotence), so let's have two endpoints and let's allow users to pick.

Reply:
Done:
- Changed content type to application/json
- Added additional upload settings endpoint
- Provide 415 status in case of  /api/ror/settings enpoint